itests/worker_test.go

```diff
@@ -438,6 +441,129 @@ func TestWindowPostWorkerDisconnected(t *testing.T) {
 	tryDl(40) // after HeartbeatInterval, the badWorker should be marked as disabled
 }
 
+type ParPostWorkerExecutor struct {
+	storiface.Storage
+
+	wait *uint32
+	cont chan struct{}
+}
+
+func (e *ParPostWorkerExecutor) GenerateWindowPoStWithVanilla(ctx context.Context, proofType abi.RegisteredPoStProof, minerID abi.ActorID, randomness abi.PoStRandomness, proofs [][]byte, partitionIdx int) (proof.PoStProof, error) {
+	atomic.AddUint32(e.wait, 1)
+	<-e.cont
+
+	return e.Storage.GenerateWindowPoStWithVanilla(ctx, proofType, minerID, randomness, proofs, partitionIdx)
+}
+
+func TestWindowPostWorkerParallel(t *testing.T) {
+	doTest := func(maxBatch, maxPart int) func(t *testing.T) {
+		return func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			_ = logging.SetLogLevel("storageminer", "INFO")
+
+			sectors := 2 * 48 * 2
```

Comment:
Could you add a comment saying how you came up with this number?

Reply:
This number is coming from that a local network has a max of 2 sectors per partition. So 2 sectors * 48 windows * 2 partitions, to be able to test that windowPoSts are processed in parallel. I can that comment if that is okay for you @magik6k ?